packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -365,20 +361,11 @@ abstract class RenderSliverScrollingPersistentHeader extends RenderSliverPersist
     return stretchOffset > 0 ? 0.0 : math.min(0.0, paintExtent - childExtent);
   }
 
-
   @override
   void performLayout() {
     final SliverConstraints constraints = this.constraints;
     final double maxExtent = this.maxExtent;
     layoutChild(constraints.scrollOffset, maxExtent);
-    final double paintExtent = maxExtent - constraints.scrollOffset;
-    geometry = SliverGeometry(
-      scrollExtent: maxExtent,
-      paintOrigin: math.min(constraints.overlap, 0.0),
-      paintExtent: clampDouble(paintExtent, 0.0, constraints.remainingPaintExtent),
-      maxPaintExtent: maxExtent,
-      hasVisualOverflow: true, // Conservatively say we do have overflow to avoid complexity.
-    );
```

Comment:
Note to future archaeologists: This code is redundant because `updateGeometry` called below also calculates and assigns `geometry`.